src/engine/strat_engine/pool.rs

```diff
@@ -207,7 +207,8 @@ impl Pool for StratPool {
         let dm = try!(DM::new());
         try!(self.thin_pool.teardown(&dm));
 
-        for bd in self.block_devs.values() {
+        let mut blockdevs = self.block_devs;
+        for (_, bd) in blockdevs.drain() {
```

Comment:
style: don't need the let binding if you change to `fn destroy(mut self)` above?

In any case 👍 I think moving self into destroy/teardown() makes intuitive sense.